src/labels/label_line.js

```diff
@@ -250,12 +251,15 @@ export default class LabelLine {
 
         switch (this.placement) {
             case PLACEMENT.CORNER:
-                position = segment[1].slice();
+                position = [
+                    Utils.roundToNearestPixel(segment[1][0]),
```

Comment:
It wouldn't make sense to snap to a pixel based unit, as this would likely be converted to some non-integer position after converting to screen space. Your shader rounding method would catch any such rational point, so it's best not to round at all, I'd think.